kratos/solving_strategies/strategies/explicit_solving_strategy.h

```diff
@@ -290,6 +290,26 @@ class ExplicitSolvingStrategy : public SolvingStrategy<TSparseSpace, TDenseSpace
         return mpExplicitBuilder;
     };
 
+    const ExplicitBuilderType& GetExplicitBuilder() const
+    {
+        KRATOS_TRY
+
+        KRATOS_DEBUG_ERROR_IF(mpExplicitBuilder == nullptr) << "The explicit builder is not initialized" << std::endl;
+        return *mpExplicitBuilder;
+
+        KRATOS_CATCH("")
+    };
+
+    ExplicitBuilderType& GetExplicitBuilder()
+    {
+        KRATOS_TRY
+
+        KRATOS_DEBUG_ERROR_IF(mpExplicitBuilder == nullptr) << "The explicit builder is not initialized" << std::endl;
+        return *mpExplicitBuilder;
+
+        KRATOS_CATCH("")
+    };
+
```

Comment:
Which changes in this PR make these necessary?

Reply:
The ExtractSolutionStepData method in the BFECC startegy. It's not critical (I could simply use the pGetter) but in general I think it's better to avoid pointers when not needed.

I actually believe that the current pGetExplicitBuilder should return a Pointer, not a reference to it. I believe that returning references to shared pointers is asking for trouble (because the reference counter does not change).

Reply:
Agree about the pointer. I'd create a `pGetExplicitBuilderAndSolver`. My point was more on the line that this is changing the public API of the base explicit strategy class, something that should be discussed in a standalone PR.

Reply:
You're right. I'll open a separate PR.